src/Components/Aspire.OpenAI/AspireOpenAIClientBuilderChatClientExtensions.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Data.Common;
+using Aspire.OpenAI;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using OpenAI;
+
+namespace Microsoft.Extensions.Hosting;
+
+/// <summary>
+/// Provides extension methods for registering <see cref="IChatClient"/> as a singleton in the services provided by the <see cref="IHostApplicationBuilder"/>.
+/// </summary>
+public static class AspireOpenAIClientBuilderChatClientExtensions
+{
+    private const string DeploymentKey = "Deployment";
+    private const string ModelKey = "Model";
+
+    /// <summary>
+    /// Registers a singleton <see cref="IChatClient"/> in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">An <see cref="AspireOpenAIClientBuilder" />.</param>
+    /// <param name="deploymentName">Optionally specifies which model deployment to use. If not specified, a value will be taken from the connection string.</param>
+    /// <param name="disableOpenTelemetry">Optional. If <see langword="true"/>, skips registering open telemetry support in the <see cref="IChatClient"/> pipeline.</param>
+    /// <returns>A <see cref="ChatClientBuilder"/> that can be used to build a pipeline around the inner <see cref="IChatClient"/>.</returns>
+    public static ChatClientBuilder AddChatClient(
+        this AspireOpenAIClientBuilder builder,
+        string? deploymentName = null,
+        bool disableOpenTelemetry = false)
+    {
+        return builder.HostBuilder.Services.AddChatClient(
+            services => CreateInnerChatClient(services, builder, deploymentName, disableOpenTelemetry));
+    }
+
+    /// <summary>
+    /// Registers a keyed singleton <see cref="IChatClient"/> in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">An <see cref="AspireOpenAIClientBuilder" />.</param>
+    /// <param name="serviceKey">The service key with which the <see cref="IChatClient"/> will be registered.</param>
+    /// <param name="deploymentName">Optionally specifies which model deployment to use. If not specified, a value will be taken from the connection string.</param>
+    /// <param name="disableOpenTelemetry">Optional. If <see langword="true"/>, skips registering open telemetry support in the <see cref="IChatClient"/> pipeline.</param>
+    /// <returns>A <see cref="ChatClientBuilder"/> that can be used to build a pipeline around the inner <see cref="IChatClient"/>.</returns>
+    public static ChatClientBuilder AddKeyedChatClient(
+        this AspireOpenAIClientBuilder builder,
+        string serviceKey,
+        string? deploymentName = null,
+        bool disableOpenTelemetry = false)
```

Comment:
How does this `disableOpenTelemetry` parameter compose with with the property on the root `OpenAISettings`?

https://github.com/dotnet/aspire/blob/ef415697283fff284e908f983b8d132d714a6630/src/Components/Aspire.OpenAI/OpenAISettings.cs#L44-L52

If I set `OpenAISettings.DisableTracing = true`, should that affect what happens here?

Reply:
I'll see if I can get the `OpenAISettings.DisableTracing` flag to control this.

That seems reasonable though it's arguably a bit clunky because the meaning doesn't align perfectly. We're never going to **disable** tracing in the sense of "block it from working". We only want a way to signal "don't auto-add the tracing middleware, and then I might add it manually in a custom place".

I think the slight mismatch of meaning isn't a big problem and if we can control this via `OpenAISettings.DisableTracing` it seems good to me. Not certain yet whether that fits the layering though.

Reply:
Done. This is definitely better. Thanks for the suggestion!